Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -1874,6 +1874,52 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
         }
     }
     
+    private func synthesizeArticleOnlyRootPage(articles: inout [DocumentationContext.SemanticResult<Article>], bundle: DocumentationBundle) {
+        let title = bundle.displayName
+        let metadataDirectiveMarkup = BlockDirective(name: "Metadata", children: [
+            BlockDirective(name: "TechnologyRoot", children: [])
+        ])
+        let metadata = Metadata(from: metadataDirectiveMarkup, for: bundle, in: self)
+        
+        if articles.count == 1 {
+            // This catalog only has one article, so we make that the root.
+            var onlyArticle = articles.removeFirst()
+            onlyArticle.value = Article(markup: onlyArticle.value.markup, metadata: metadata, redirects: onlyArticle.value.redirects, options: onlyArticle.value.options)
+            registerRootPages(from: [onlyArticle], in: bundle)
+        } else if let nameMatchIndex = articles.firstIndex(where: { $0.source.deletingPathExtension().lastPathComponent == title }) {
```

Comment:
This is what takes to add the "index" as possible root, + the unit test ofc.

```suggestion
        } else if let nameMatchIndex = articles.firstIndex(where: { $0.source.deletingPathExtension().lastPathComponent == title || $0.source.deletingPathExtension().lastPathComponent == "index" })  {
            // This catalog has an article with the same name as the catalog itself, or an article named index, so we make that the root.
```

Reply:
Yes, it's very easy to implement another file name as a possible root but I'm not convinced there's a strong use case for it. If people want their "index.md" file to be a root for their article-only documentation they can always add a `@TechnologyRoot` directive to it. 

What—in my opinion—makes the file matching the catalog name different is that that if we didn't turn that into a root page it would result in a "/documentation/CatalogName/CatalogName" URL (once for the catalog name and once for the article name) that I think people would find mildly surprising.

I [searched for usages of "index.md" within documentation catalogs][search] but almost all of the matches are for files that are the documentation extension of the module, so this wouldn't apply to those anyway. A [refined search][search2] show 5 projects that have an "Index.md" / "index.md" file that isn't the documentation extension for the module.

[search]: https://github.com/search?q=path%3A%2F%5E*%5C.docc%5C%2Findex%5C.md%2F+%22%23+%22+language%3AMarkdown&type=code&p=3
[search2]: https://github.com/search?q=path%3A%2F%5E*%5C.docc%5C%2Findex%5C.md%2F+%2F%5E%23+%5B%5E%60%5D%2F+language%3AMarkdown+NOT+is%3Afork&type=code

- "Apollo" in "apollographql"
- "hummingbird-docs" in "hummingbird-project"

Already have `@TechnologyRoot` directives but those catalog also has a module so any root page file name logic wouldn't apply.

- "Conformance" in "swift-protobuf"
- "SwiftyESBuild" in "tuist"
- "leetcode" in "rossmassey"

have modules in their catalogs so this root page file name logic wouldn't apply.